frigate/comms/events_updater.py

```diff
@@ -1,100 +1,51 @@
 """Facilitates communication between processes."""
 
-import zmq
-
 from frigate.events.types import EventStateEnum, EventTypeEnum
 
-SOCKET_PUSH_PULL = "ipc:///tmp/cache/events"
-SOCKET_PUSH_PULL_END = "ipc:///tmp/cache/events_ended"
+from .zmq_proxy import Publisher, Subscriber
 
 
-class EventUpdatePublisher:
+class EventUpdatePublisher(Publisher):
     """Publishes events (objects, audio, manual)."""
 
+    topic_base = "event/"
+
     def __init__(self) -> None:
-        self.context = zmq.Context()
-        self.socket = self.context.socket(zmq.PUSH)
-        self.socket.connect(SOCKET_PUSH_PULL)
+        super().__init__("update")
 
     def publish(
         self, payload: tuple[EventTypeEnum, EventStateEnum, str, dict[str, any]]
     ) -> None:
-        """There is no communication back to the processes."""
-        self.socket.send_pyobj(payload)
-
-    def stop(self) -> None:
-        self.socket.close()
-        self.context.destroy()
+        super().publish(payload)
 
 
-class EventUpdateSubscriber:
+class EventUpdateSubscriber(Subscriber):
     """Receives event updates."""
 
-    def __init__(self) -> None:
-        self.context = zmq.Context()
-        self.socket = self.context.socket(zmq.PULL)
-        self.socket.bind(SOCKET_PUSH_PULL)
-
-    def check_for_update(
-        self, timeout=1
-    ) -> tuple[EventTypeEnum, EventStateEnum, str, dict[str, any]]:
-        """Returns events or None if no update."""
-        try:
-            has_update, _, _ = zmq.select([self.socket], [], [], timeout)
-
-            if has_update:
-                return self.socket.recv_pyobj()
-        except zmq.ZMQError:
-            pass
+    topic_base = "event/"
 
-        return None
-
-    def stop(self) -> None:
-        self.socket.close()
-        self.context.destroy()
+    def __init__(self) -> None:
+        super().__init__("update")
 
 
-class EventEndPublisher:
+class EventEndPublisher(Publisher):
     """Publishes events that have ended."""
 
+    topic_base = "event/"
+
     def __init__(self) -> None:
-        self.context = zmq.Context()
-        self.socket = self.context.socket(zmq.PUSH)
-        self.socket.connect(SOCKET_PUSH_PULL_END)
+        super().__init__("ended")
```

Comment:
It's a little confusing, but here is when these messages get sent and what they contain:

The update publisher publishes all event updates based on object processing. This includes start/update/end.
The event end publisher is based on the event maintainer and gets published only after the finished event is saved to the DB (or not).

I didn't want to use end here because that can be confused with the EventUpdate end.

I'm ok to change to "updated" for the other one though.

Reply:
maybe we can change `ended` to `finalized`, `cleared`, or something